pkg/controllers/binding/binding_controller.go

```diff
@@ -82,9 +82,11 @@ func (c *ResourceBindingController) Reconcile(ctx context.Context, req controlle
 
 	if !binding.DeletionTimestamp.IsZero() {
 		klog.V(4).Infof("Begin to delete works owned by binding(%s).", req.NamespacedName.String())
-		if err := helper.DeleteWorks(c.Client, req.Namespace, req.Name, binding.Labels[workv1alpha2.ResourceBindingPermanentIDLabel]); err != nil {
-			klog.Errorf("Failed to delete works related to %s/%s: %v", binding.GetNamespace(), binding.GetName(), err)
-			return controllerruntime.Result{}, err
+		if binding.Spec.OrphaningDeletion == policyv1alpha1.OrphaningDeletionNever || binding.Spec.OrphaningDeletion == policyv1alpha1.OrphaningDeletionComplete {
```

Comment:
I'd suggest using a `switch` here, as it will make the logic clearer. It will also make it easier for us to add comments for each `case`.

Reply:
> I'd suggest using a switch here, as it will make the logic clearer. It will also make it easier for us to add comments for each case.

Only one case now, it's not necessary